### PR TITLE
Implement Add and Delete deliverable features

### PR DESCRIPTION
**Pull request by gabztcr**
-Amend the attributes under deliverable
-Create new java classes for each attribute
-Amend UI files for deliverable
-Implement add deliverable feature
-Implement delete deliverable feature

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/deliverable/ParserUtil.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    /**
+     * Parses a {@code String deadline} into a {@code Deadline}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code deadline} is invalid.
+     */
+    public static Deadline parseDeadline(String deadline) throws ParseException {
+        requireNonNull(deadline);
+        if (deadline.equals("NIL")) {
+            return new Deadline("00-00-0000 00:00");
```
Comment (shadowezz) [inline src/main/java/seedu/address/logic/parser/deliverable/ParserUtil.java:74]: Maybe its better to display NIL like the rest

**Review by shadowezz — COMMENTED** [review]
LGTM other than some minor typos and suggestions

File: src/main/java/seedu/address/model/deliverable/deliverable/Deadline.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * Represents a Deliverable's deadline in the deliverable book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDeadline(String)}
+ */
+public class Deadline {
+
+    public static final String MESSAGE_CONSTRAINTS = "Deadlines must be in DD-MM-YYYY HH:MM format";
+
+    public static final String VALIDATION_REGEX = "[0-9][0-9]-[0-9][0-9]-[0-9][0-9][0-9][0-9] [0-9][0-9]:[0-9][0-9]";
+
+    public final String value;
```
[the remark this answers is not in the document]
Comment (gabztcr) [reply src/main/java/seedu/address/model/deliverable/deliverable/Deadline.java:16]: I made two fields for Deadline class. One a String, and another a LocalDateTime. We can use the latter for comparison, and leave the former as it is for string rendering (y)

**Comment by gabztcr** [comment]
Checks failing due to Edit command not available yet.

**Comment by gabztcr** [comment]
Fixes #3 and #5

**Pull request merged**